hoodie-common/src/main/java/com/uber/hoodie/common/util/FSUtils.java

```diff
@@ -254,6 +313,53 @@ public static String getBaseCommitTimeFromLogPath(Path path) {
     return matcher.group(2);
   }
 
+  /**
+   * Get TaskId used in log-path
+   */
+  public static Integer getTaskPartitionIdFromLogPath(Path path) {
+    Matcher matcher = LOG_FILE_PATTERN.matcher(path.getName());
+    if (!matcher.find()) {
+      throw new InvalidHoodiePathException(path, "LogFile");
+    }
+    String val = matcher.group(7);
+    return val == null ? null : Integer.parseInt(val);
+  }
+
+  /**
+   * Get Write-Token used in log-path
+   */
+  public static String getWriteTokenFromLogPath(Path path) {
+    Matcher matcher = LOG_FILE_PATTERN.matcher(path.getName());
+    if (!matcher.find()) {
+      throw new InvalidHoodiePathException(path, "LogFile");
```

Comment:
how do you intend to handle existing logs that don't have a write token? log writer picks up `writeToken` passed in `orElse` above and writes to a new log file?

Reply:
Changed the behavior such that If there is an existing log-file without write-token, we will first rollover to new log version and then start writing.  Also, added consistent ordering of log-files covering cases where there is same log-version but different write-token.